cpp/src/arrow/util/value_parsing.cc

```diff
@@ -14,6 +14,8 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+#ifndef FASTFLOAT_ALLOWS_LEADING_PLUS
```

Comment:
Why do we need this `ifndef`? What situation are you thinking? I think that this is needless...